src/sync-pending-events.ts

```diff
@@ -255,4 +256,43 @@ async function getLatestBlockNumber(provider: string) {
     }
   });
   return latestBlockNumber
+}
+
+async function getValidBlockNumber(provider: string, startPoint: number, endPoint: number) {
+  if(endPoint < startPoint){
+    console.log('ISSUE: LatestBlockNumber Get from RpcSource is smaller than real value, we will not change lastCheckedBlockNumber');
+    return startPoint
+  }
+  let web3 = getWeb3FromSource(provider);
+  let chekced =  false;
+  let BlockNumberIssue = false;
+  while(!chekced){
+    await web3.eth.getBlock(`${endPoint}`).then(async block => {
+      if (block == null) {
+        let [lowerBoundary, upperBoundary] = await binarySearchValidBlock(provider, startPoint, endPoint);
+        startPoint = lowerBoundary;
+        endPoint = upperBoundary;
+        BlockNumberIssue = true;
+      }else {
+        if (BlockNumberIssue){
+          console.log('ISSUE: LatestBlockNumber Get from RpcSource is Larger than Real Value, the actual lastCheckedBlockNumber is: ' + endPoint);
```

Comment:
This comment is not accurate, maybe it is not because of "Larger", maybe just say "Cannot find exactly block from block number: endPoint + 1, the actual lastCheckedBlockNumber is: endPoint"？